### ISSUE TITLE
[Build] Windows CUDA build fails due to moderngpu

### ISSUE THREAD

**Issue by BarclayII** [dmlc/dgl#651]
## 🐛 Bug

In `third_party/minigun/third_party/moderngpu/src/moderngpu/tuple.hxx:226-256` one has the following declaration which used GNU extension `__attribute__((aligned))`:
```C++
template<typename... args_t>
struct tuple : detail::tuple_impl<0, args_t...> { 
//...
} __attribute__((aligned));
```

which is not recognized by Visual C++ compiler (hence NVCC on Windows).

EDIT: full error message:
```
  c:\programdata\miniconda3\conda-bld\dgl_1560355307264\work\third_party\minigun\third_party\moderngpu\src\moderngpu\tuple.hxx(256): error : expected a ";" [%SRC_DIR%\build\dgl.vcxproj]
  c:\programdata\miniconda3\conda-bld\dgl_1560355307264\work\third_party\minigun\third_party\moderngpu\src\moderngpu\launch_params.hxx(70): error : expected a ")" [%SRC_DIR%\build\dgl.vcxproj]
```

**Comment by yzh119** [comment]
Do you know how to solve this? If so we could fork moderngpu and make it a dependency of minigun.

**Comment by jermainewang** [comment]
I'm on it. Currently, moderngpu is only required when the given frontier is partial. Since DGL only uses full frontier, we could safely remove that dependency.

**Comment by jermainewang** [comment]
Should be fixed in #652. Please close if no further issue.
